docs/specifications/cva6_requirement_specification.adoc

```diff
@@ -0,0 +1,391 @@
+
+== Reference specifications
+
+[RVcmo] “RISC-V Base Cache Management Operation ISA Extensions, version 1.0-fd39d01, 2022-01-12” +
+
+[CLINT] Core-Local Interruptor (CLINT), “SiFive E31 Core Complex Manual v2p0”, chapter 6, https://static.dev.sifive.com/SiFive-E31-Manual-v2p0.pdf[https://static.dev.sifive.com/SiFive-E31-Manual-v2p0.pdf] + 
+
+== Other documents
+
+[FPGAreset] Ken Chapman, “Get Smart About Reset: Think Local, Not Global”, Xilinx WP272 white paper, https://www.xilinx.com/support/documentation/white_papers/wp272.pdf[https://www.xilinx.com/support/documentation/white_papers/wp272.pdf]. +
+
+== Functional requirements
+
+=== General requirement
+
+|===
+
+|GEN&#8209;10|CVA6 shall be *fully compliant with RISC-V specifications* [RVunpriv], [RVpriv] and [RVdbg] by implementing all mandatory features for the set of extensions that are selected and by passing [RVcompat] compatibility tests.
+
+|===
+
+As the RISC-V specification leaves space for options, this specification is a place where to specify these options.
+
+=== RISC-V standard instructions
+
+To ease tracing to verification, the extensions have been split in independent requirements.
+|===
+
+|ISA&#8209;10|CV64A6 shall support *RV64I* base instruction set, version 2.1.
+|ISA&#8209;20|CV32A6 shall support *RV32I* base instruction set, version 2.1.
+|ISA&#8209;30|CVA6 shall support the **M** extension (integer multiply and divide), version 2.0.
+|ISA&#8209;40|CVA6 shall support the **A** extension (atomic instructions), version 2.1.
+|ISA&#8209;50|CV32A6 shall support as an *option* the **F** extension (single-precision floating-point), version 2.2.
+|ISA&#8209;60|CV64A6 shall support as an *option* the **F** and **D** extensions (single- and double-precision floating-point), version 2.2.
+|ISA&#8209;70|CV64A6 shall support as an *option* the **F** extension (single-precision without double-precision floating-point), version 2.2.
+|ISA&#8209;80|CVA6 shall support the **C** extension (compressed instructions), version 2.0 as an *option*.
+|ISA&#8209;90|CVA6 shall support the *Zicsr* extension (CSR instructions), version 2.0.
+|ISA&#8209;100|CVA6 shall support the *Zifencei* extension, version 2.0.
+|ISA&#8209;110|As an *option*, the duration of instructions shall be independent from the operand values.
+|===
+
+Note to ISA-110: In the current design, the duration of the division is data-dependent, which can be a security issue.
+
+=== Privileges and virtual memory
+
+The MMU includes a TLB and a hardware PTW.
+
+|===
+
+|PVL&#8209;10|CVA6 shall support *machine*, *supervisor***,** *user* and *debug* privilege modes.
+|PVL&#8209;20|CV64A6 shall support as an *option* the *Sv39* virtual memory, version 1.11.
+|PVL&#8209;30|CV32A6 shall support as an *option* the *Sv32* virtual memory version 1.11.
+|PVL&#8209;40|CVA6 instances that do not feature virtual memory shall support the *Bare* mode.
```

Comment:
Not clear if this is a direct consequence of not supporting virtual memory or is Bare mode an optional feature that might or might not be turned on?

Reply:
This is indeed a direct consequence as the virtual memory is an option. I marked a requirement for it so that we do not forget to verify the Bare mode.